mlcomp/parallelm/pipeline/component_info.py

```diff
@@ -0,0 +1,319 @@
+                ret[json_fields.COMPONENT_DESC_PROGRAM_FIELD] = obj.program
+            if obj.component_class:
+                ret[json_fields.COMPONENT_DESC_CLASS_FIELD] = obj.component_class
+            if obj.model_behavior:
+                ret[json_fields.COMPONENT_DESC_MODEL_BEHAVIOR_FIELD] = obj.model_behavior
+            if obj.use_mlops:
+                ret[json_fields.COMPONENT_DESC_USE_MLOPS_FIELD] = obj.use_mlops
+            if obj.deps:
+                ret[json_fields.COMPONENT_DESC_PYTHON_DEPS] = obj.deps
+            if obj.include_glob_patterns:
+                ret[json_fields.COMPONENT_DESC_INCLUDE_GLOB_PATTERNS] = obj.include_glob_patterns
+            if obj.exclude_glob_patterns:
+                ret[json_fields.COMPONENT_DESC_EXCLUDE_GLOB_PATTERNS] = obj.exclude_glob_patterns
+            if obj.inputs:
+                ret[json_fields.COMPONENT_DESC_INPUT_INFO_FIELD] = obj.inputs
+            if obj.outputs:
+                ret[json_fields.COMPONENT_DESC_OUTPUT_INFO_FIELD] = obj.outputs
+            if obj.arguments:
+                ret[json_fields.COMPONENT_DESC_ARGUMENTS] = obj.arguments
+            return ret
+        elif isinstance(obj, ComponentConnectionInfo):
+            ret = OrderedDict()
+            if obj.description:
+                ret[json_fields.CONNECTION_DESC_DESCRIPTION_FIELD] = obj.description
+            if obj.label:
+                ret[json_fields.CONNECTION_DESC_LABEL_FIELD] = obj.label
+            if obj.default_component:
+                ret[json_fields.CONNECTION_DESC_DEFAULT_COMPONENT_FIELD] = obj.default_component
+            if obj.type:
+                ret[json_fields.CONNECTION_DESC_TYPE_FIELD] = obj.type
+            if obj.group:
+                ret[json_fields.CONNECTION_DESC_GROUP_FIELD] = obj.group
+            return ret
+        elif isinstance(obj, ComponentArgumentInfo):
+            ret = OrderedDict()
+            if obj.key:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_KEY] = obj.key
+            if obj.type:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_TYPE] = obj.type
+            if obj.label:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_LABEL] = obj.label
+            if obj.description:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_DESCRIPTION] = obj.description
+            if obj.env_var:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_ENV_VAR] = obj.env_var
+            if obj.optional:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_OPTIONAL] = obj.optional
+            if obj.tag:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_TAG] = obj.tag
+            if obj.default_value:
+                ret[json_fields.COMPONENT_DESC_ARGUMENT_DEFAULT_VAL] = obj.default_value
+            return ret
+        else:
+            # Let the base class default method raise the TypeError
+            return json.JSONEncoder.default(self, obj)
+
+
+class ComponentInfo(object):
```

Comment:
Maybe at a larger scope - we better have a single source of truth! Either we generate this kind of class from a yaml/json template that provides the schema for a component and in addition all component's json are validated against this template, or we'll have a python class that generates a component's info schema and again all component's json should be validated against this generated template. WDYT?